### PR TITLE
Various fixes

### PR DESCRIPTION
**Pull request by melekes**
Fixes https://github.com/tendermint/tendermint/issues/4136
Fixes https://github.com/tendermint/tendermint/issues/4166

* [x] Referenced an issue explaining the need for the change
* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [ ] Wrote tests
* [x] Updated CHANGELOG_PENDING.md

### CODE REVIEW

**Comment by marbar3778** [comment]
do you want to add some extra changelog entries with what https://github.com/tendermint/tendermint/issues/4166 recommends? then we can close that one as well

**Comment by melekes** [comment]
> do you want to add some extra changelog entries with what #4166 recommends? then we can close that one as well

will do

**Comment by tessr** [comment]
This PR looks like it addresses a number of separate and unrelated issues. Can they be split into multiple PRs/commits, please? This will make debugging easier in the future, if we ever have to go git history spelunking 😇

**Comment by melekes** [comment]
What about merge commit?

File: tools.mk [hunk 1]
```diff
@@ -59,28 +59,45 @@ check_tools:
 certstrap: $(CERTSTRAP)
 $(CERTSTRAP):
 	@echo "Get Certstrap"
-	@go get github.com/square/certstrap@338204a88c4349b1c135eac1e8c14c693ad007da
+	@go get github.com/square/certstrap@v1.2.0
+
+# Choose protobuf binary based on OS.
+PROTOC_ZIP=""
+ifneq ($(OS),Windows_NT)
```
Comment (melekes) [inline tools.mk]: should we try to make this work for Windows || 32 bit systems || ARM?

**Comment by tessr** [comment]
> What about merge commit?

I'm not sure we're talking about the same thing--I mean that each commit should represent an independent change, so that we can easily identify the motivation for each change. And the right way to do that, given our git flow, is probably to open a separate PR for each one.

File: docs/spec/blockchain/blockchain.md [hunk 2]
```diff
@@ -187,6 +187,8 @@ type DuplicateVoteEvidence struct {
 }
 ```
 
+Votes are lexicographically sorted on `BlockID`.
```
Comment (marbar3778) [inline docs/spec/blockchain/blockchain.md]: this should be added to the spec repo. the specs in this repo are going to be removed so we dont have duplicates

**Comment by melekes** [comment]
> I mean that each commit should represent an independent change, so that we can easily identify the motivation for each change

https://github.com/tendermint/tendermint/pull/4170/commits look pretty good to me

> I'm not sure we're talking about the same thing

Sorry. I was trying to say that we can "Merge" this PR with merge commit (instead of "Squash and merge"). That way we will a) preserve independent commit history b) spare my time. What do you think?

**Comment by tessr** [comment]
Oh yeah, that's fine. But let's rebase onto master so we don't leave a merge commit. (And speaking of which, can you rebase this branch to remove the merge commit that's already on it?)

File: docs/spec/blockchain/blockchain.md (hunk 2 above)
Comment (marbar3778) [reply docs/spec/blockchain/blockchain.md]: added to the spec repo

File: tools.mk [hunk 3]
```diff
@@ -59,28 +59,54 @@ check_tools:
...
+# Choose protobuf binary based on OS (only works for 64bit Linux and Mac).
+# NOTE: On Mac, installation via brew (brew install protoc) might be favorable.
+PROTOC_ZIP=""
+ifneq ($(OS),Windows_NT)
+		UNAME_S := $(shell uname -s)
+		ifeq ($(UNAME_S),Linux)
+			PROTOC_ZIP="protoc-3.10.1-linux-x86_64.zip"
+		endif
+		ifeq ($(UNAME_S),Darwin)
+				PROTOC_ZIP="protoc-3.10.1-osx-x86_64.zip"
```
Comment (melekes) [inline tools.mk]: formatting is off

**Pull request merged**